assembl/static2/js/app/components/login/signupForm.jsx

```diff
@@ -107,16 +114,44 @@ class SignupForm extends React.Component<void, SignupFormProps, SignupFormState>
         <div className="box">
           <form className="signup" onSubmit={this.signupHandler}>
             {textFields &&
-              textFields.map(tf => (
-                <FormGroup key={tf.id}>
-                  <FormControl
-                    type={tf.fieldType.toLowerCase()}
-                    name={tf.identifier === 'CUSTOM' ? tf.id : tf.identifier.toLowerCase()}
-                    placeholder={tf.title}
-                    onChange={this.handleInput}
-                  />
-                </FormGroup>
-              ))}
+              textFields.map((field) => {
+                if (field.__typename === 'TextField') {
+                  return (
+                    <FormGroup key={field.id}>
+                      <FormControl
+                        type={field.fieldType.toLowerCase()}
+                        name={field.identifier === 'CUSTOM' ? field.id : field.identifier.toLowerCase()}
+                        placeholder={field.required ? `${field.title}*` : field.title}
+                        onChange={this.handleInput}
+                        required={field.required}
+                      />
+                    </FormGroup>
+                  );
+                }
+
+                if (field.__typename === 'SelectField' && field.options) {
```

Comment:
There is a code smell with this check. The `SelectField` model hasn't made an option mandatory. So, an empty select field was selected with no option wouldn't render. The impact is non-deterministic as the fields could be very important, but they aren't rendered for use, or non-important fields just won't render.

Reply:
It will render and empty select.